### PR TITLE
feat: calculate body class names seperately

### PR DESCRIPTION
**Pull request by Raubzeug**
Calculate base body class names in one place and export function that allows to get basic body classes with theme to use them in SSR.

### CODE REVIEW

File: src/components/theme/getBodyClassName.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import type {RealTheme} from './types';
+import type {BodyClassNameModifiers} from './updateBodyClassName';
+
+const ROOT_CLASS_NAME = 'root';
+
+const bNew = blockNew(ROOT_CLASS_NAME);
+const b = block(ROOT_CLASS_NAME);
+
+export function getDeprecatedBodyRootClassName(
+    modifier?: Partial<BodyClassNameModifiers & {theme: RealTheme | boolean}>,
```
Comment (ogonkov) [inline src/components/theme/getBodyClassName.ts:12]: I'm not sure that `theme` could be a bool

Comment (ogonkov) [reply src/components/theme/getBodyClassName.ts:12]: Oh, it is, sorry

File: src/components/theme/getBodyClassName.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+) {
+    return b(modifier);
+}
+export function getBodyRootClassName(
```
Comment (amje) [inline src/components/theme/getBodyClassName.ts:16]: "body" is redundant here, root can be on any element

File: src/components/theme/getBodyClassName.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    modifier?: Partial<BodyClassNameModifiers & {theme: RealTheme | boolean}>,
+    addition?: string[],
+) {
+    return bNew(modifier, addition);
+}
+
+export function getBaseBodyRootClassName(theme?: RealTheme) {
```
Comment (amje) [inline src/components/theme/getBodyClassName.ts:23]: Suggestion: `getInitialRootClassName`

**Review by amje — CHANGES_REQUESTED** [review]

**Review by ogonkov — APPROVED** [review]

**Review by amje — APPROVED** [review]

**Review by korvin89 — APPROVED** [review]

**Pull request merged**